### ISSUE TITLE
Metals config is not being picked up

### ISSUE THREAD

**Issue by zetashift** [scalameta/nvim-metals#201]
So I'm using Fennel, and I'm settings some options like `fallbackScalaVersion` or `showInferredTypes` however `:MetalsInfo` does not pick these up, and I'm not sure why, because this even happens with a minimal config.

Also the LSP trace:

```json
[Trace - 06:49:43 PM] Received notification 'workspace/didChangeConfiguration'
Params: {
  "settings": {
    "metals": {
      "superMethodLensesEnabled": true,
      "showInferredType": true,
      "showImplicitArguments": true,
      "fallbackScalaVersion": "3.0.1"
    }
  }
}
```

**To Reproduce** Steps to reproduce the behavior:

My config is here: https://github.com/zetashift/.dotfiles/blob/main/.config/nvim/fnl/magic/plugin/metals.fnl 

But even with the starting config (adding some config options like fallbackScalaVersion): https://github.com/scalameta/nvim-metals/discussions/39 , I didn't see anything in `:MetalsInfo` 

**Expected behavior**

The settings being picked up correctly

**Installation:**

- Operating system: Linux
- Nvim version: NVIM v0.6.0-dev+135-g222cd4322
- Commit of nvim-metals: bcd8a28f8b647a21adba2d0acd777b742e478d8b
- Metals version: 0.10.6-M1

**Additional context**

This is the generated Lua from my config:

```lua
_G.metals_init = function()
  local metals = require("metals")
  local conf = metals.bare_config
  conf["init_options"] = {statusBarProvider = "on"}
  conf["settings"] = {fallbackScalaVersion = "3.0.1", showImplicitArguments = true, showInferredType = true, superMethodLensesEnabled = true}
  return metals.initialize_or_attach(conf)
end
vim.cmd(("augroup " .. "lsp" .. "\nautocmd!"))
nvim.ex.autocmd("FileType scala,sbt :call v:lua.metals_init()")
vim.cmd("augroup END")

```

**Search terms**

config metals

**Comment by ckipp01** [comment]
Thanks for reporting @zetashift. So I finally see what is going wrong here. Recently I switched from the `M = {}` semantics to exporting individual functions and I didn't take into account that by me doing that it was screwing up `metals.lua` accessing my cached values. So while the settings are being set correctly in your above case, it was an issue of the info command not being able to access the settings. This should be fixed in #202, so let me know if that doesn't fix the issue for you.

**Comment by zetashift** [comment]
![image](https://user-images.githubusercontent.com/1857826/129492032-755c1767-bdb9-48d7-8f68-fe6273260b7c.png)

Thanks!! You're the best :D.
